crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -367,17 +371,33 @@ pub(crate) fn do_slash_operator<T: Config>(
             let locked_amount = DepositOnHold::<T>::take((operator_id, nominator_id.clone()));
 
             // convert any previous epoch deposits
-            do_convert_previous_epoch_deposits::<T>(operator_id, &mut deposit)?;
+            match do_convert_previous_epoch_deposits::<T>(
+                operator_id,
+                &mut deposit,
+                current_domain_epoch_index,
+            ) {
+                // Share price may be missing if there is deposit happen in the same epoch as slash
+                Ok(()) | Err(TransitionError::MissingOperatorEpochSharePrice) => {}
```

Comment:
Similar question here.

Reply:
Similar to de-register, deposit/withdraw then slash in the same epoch is allowed while slash then deposit/withdraw is not, and after slash the operator is remove from the domain's `next_operators` and there won't be share price inserted for this operator.

It seems there is no test case to cover this path, I will add one to ensure if we propagate the `MissingOperatorEpochSharePrice` error here it will fail.